core/src/main/java/org/jdbi/v3/core/mapper/reflect/BeanMapper.java

```diff
@@ -32,7 +32,7 @@
  *
  * The mapped class must have a default constructor.
  */
-public class BeanMapper<T> extends PojoMapper<T> {
+public final class BeanMapper<T> extends PojoMapper<T> {
```

Comment:
I do like the philosophy that kotlin has: Anything that is not explicitly designed and labeled as extensible should not be extended. I don't think there is value in extending the bean mapper because it is just an extension on the pojo mapper.  

I agree with the mocking but then again, no one should (?) mock a mapper.